dockstore-event-consumer/src/main/java/io/dockstore/consumer/EventConsumer.java

```diff
@@ -0,0 +1,133 @@
+/*
+ *    Copyright 2017 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.dockstore.consumer;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import com.amazonaws.services.sqs.model.DeleteMessageRequest;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+import io.dockstore.common.model.BasicMessage;
+import io.dockstore.consumer.handler.DOIHandler;
+import io.dockstore.consumer.handler.MessageHandler;
+import org.apache.commons.configuration2.PropertiesConfiguration;
+import org.apache.commons.configuration2.builder.fluent.Configurations;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class EventConsumer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(EventConsumer.class);
+
+    /**
+     * the maximum wait time allowed
+     */
+    private static final int MAX_WAITTIME_SECONDS = 20;
```

Comment:
Wait time possibly separated by underscore unless there's a limit to # of underscores